packages/runtime-core/src/componentRenderUtils.ts

```diff
@@ -318,8 +319,8 @@ export function updateHOCHostEl(
   { vnode, parent }: ComponentInternalInstance,
   el: typeof vnode.el // HostNode
 ) {
-  while (parent && parent.subTree === vnode) {
-    ;(vnode = parent.vnode).el = el
+  while (parent && isSameVNodeType(parent.subTree, vnode)) {
+    ;(vnode = parent.vnode).el = parent.subTree.el = el
     parent = parent.parent
```

Comment:
The vnode which render from keep-alvie maybe is an copy.Thus the vnode isn't euquals to `parent.subTree`.
And the same as need update the `el` of `parent.subTree`.